### PR TITLE
Multiple frame detection and injection

### PR DESCRIPTION
**Pull request by JCCR**
This is part of the ongoing development of multiple iframe document support.

These changes add a way for the `CrossFrame` module to detect other frames on the page and inject hypothesis client code as an embed script in these frames. Once a frame has been injected the existing code will go on and do a handshake (`Discovery`) to establish communication between the frames (`Bridge`). This works under the assumption that the frames will be cross domain accessible in some way.

Merging in these changes will not activate the new code paths because of a "feature flag" that needs to be enabled. You can give the changes a try by setting the `MULTI_FRAME_SUPPORT=true` environment variable in your shell for development, or by using the `enableMultiFrameSupport: true` key/value pair in a hypothesis client config object.

### CODE REVIEW

**Comment by JCCR** [comment]
@sean-roberts 
Updated the description, added testing

**Comment by JCCR** [comment]
Rebased & removed the incomplete implementation on the Sidebar side that handled iframe destruction. Better implementation for that coming in a separate PR

File: scripts/gulp/live-reload-server.js [hunk 1]
```diff
@@ -33,43 +42,103 @@ function LiveReloadServer(port, config) {
...
+                window.location.reload();
+              }
+            });
+      
+            var embedScript = document.createElement('script');
+            embedScript.src = '${config.clientUrl}';
+            document.body.appendChild(embedScript);
+            
+            var iframeIsAdded = false;
+            document.querySelector('#add-test').addEventListener('click', function() {
```
Comment (sean-roberts) [inline scripts/gulp/live-reload-server.js]: document.querySelector('#add-test') is null when not in iframe test mode - causing this line to throw
